crates/solver/src/driver.rs

```diff
@@ -435,20 +441,6 @@ impl Driver {
             };
             if let Some(hash) = hash {
                 tracing::debug!(?hash, "settled transaction");
```

Comment:
suggestion: At this point we might as well drop `if let Some(hash)...` and just put specific log statements into each `match` arm.